src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +284,165 @@ def get_info(
         self, request: InfoRequest, context: grpc.ServicerContext
     ) -> InfoResponse:
         return InfoResponse(**info_response_kv_args())
+
+    def _execute_task_in_thread(
+        self, run_request: RunTaskRequest, task_execution: _TaskExecution
+    ):
+        response = RunTaskResponse(
```

Comment:
Isn't this variable initialization redundant? It's set on all code paths below.
I usually do in this case:
```
response: Optional[RanTaskResponse] = None
```

Reply:
It's not set on all code paths, but it's not obvious, which is why I like to ensure sensible values for all variables that are accessed in the exception path before the exception path could be invoked.

Reply:
What's the reason of having code paths that might not return a value that represents results of this code path?
Such code is much harder for code reader to understand.

Reply:
Done